tests/cases/string/lower.test

```diff
@@ -0,0 +1,15 @@
+### SUBSTRAIT_SCALAR_TEST: v1.0
+### SUBSTRAIT_INCLUDE: '/extensions/functions_string.yaml'
+
+# basic: Basic examples without any special cases
+lower('ABC'::str) = 'abc'::str
+lower('aBc'::str) = 'abc'::str
+lower('abc'::str) = 'abc'::str
+lower(''::str) = ''::str
+
```

Comment:
future:  we are going to need to revisit all of the string cases once we add collation and character set.  (For instance, the capital letter I in Turkish does not become i but a version without a dot.)